test/cli_command_test/test_role.py

```diff
@@ -101,6 +103,20 @@ def test_wrong_options_syspurpose_role(self):
         except SystemExit as e:
             self.assertEqual(e.code, os.EX_USAGE)
 
+    def test_username_on_registered_system(self):
+        self.cc.is_registered = Mock(return_value=True)
+        self.cc.options = Mock()
+        self.cc.options.set = None
+        self.cc.options.unset = None
+        self.cc.options.to_add = None
+        self.cc.options.to_remove = None
+        self.cc.options.show = None
+        self.cc.options.username = "admin"
```

Comment:
I would expect to see here `self.cc.options.list = True`.

Reply:
The new code in `_validate_options()` does not check if the `--list` argument is set or not. How is this different from setting it to `None` or not having it specified at all?

Reply:
That's true, but we try to mimic role command with `--list` option as much as possible. It is possible that somebody will add check to `_validate_option()` if the `--list` option is set or not in the future. In that case your unit test could give false positive result. When you are coding something it is also important to look into the future and consider future risks not only current state.